### DISCUSSION TITLE
base64 encode or decode cookie values

### DISCUSSION THREAD

**Discussion by WolfSoko** [Moustachauve/cookie-editor#130] [Ideas]
Cookies are typically base64 or URL encoded. As a software developer, displaying the decoded value would significantly enhance efficiency. Moreover, having the capability to swiftly encode a provided value would be beneficial.

**Comment by Moustachauve** [comment]
Are you thinking about something that would detect Base64 and show the decoded version next to it?
Kinda like this quick sketch I just did:
![image](https://github.com/Moustachauve/cookie-editor/assets/2206577/dd4565aa-93c6-4920-a01a-2897c1cac93e)

Doing automatic decoding like this sound very easy. I could also let you modify the value of the decoded text and get it re-encoded automatically. I'm not sure how I would make the UI work for encoding a value out of nowhere to base64 without it being too much in the way.

I think doing it contextually is the best way for UX but it could be hard to "guess" :) (Maybe I can implement shortcuts, like "@base64"

**Comment by WolfSoko** [comment]
Yes, maybe start with that as a first iteration and see how it works out. 
I also think that Base64 and URL enoding can be
detected easily.

**Reply by WolfSoko** [reply]
<img width="377" alt="image" src="https://github.com/Moustachauve/cookie-editor/assets/8589105/1dbf1f8a-8b8d-4a7b-b650-e3ea5583274c">

For an first iteration something like that with small klickable links under the value field could be helpfull. When the user clicks,  just replacing the shown value of the cookie inside the value field
